maas/resource_maas_boot_source.go

```diff
@@ -1,12 +1,126 @@
 package maas
 
 import (
+	"context"
 	"fmt"
+	"strconv"
 
 	"github.com/canonical/gomaasclient/client"
 	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+func resourceMAASBootSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMaasBootSourceRead,
+		Description: "Provides a resource to fetch the MAAS boot source.",
+
+		Schema: map[string]*schema.Schema{
+			"created": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The creation time of the boot source.",
+			},
+			"keyring_data": {
```

Comment:
`keyring_data` and `keyring_filename` are mutually exclusive. Please take a look at `ConflictsWith` and how it can be used. This is an example from the MAAS provider code:

- https://github.com/canonical/terraform-provider-maas/blob/master/maas/resource_maas_vm_host.go#L108
- https://github.com/canonical/terraform-provider-maas/blob/master/maas/resource_maas_vm_host.go#L175